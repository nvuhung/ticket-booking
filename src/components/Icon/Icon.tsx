import React from 'react';

import { ReactComponent as ZoomInIcon } from '../../assets/svg/ic-zoom-in.svg';
import { ReactComponent as ZoomOutIcon } from '../../assets/svg/ic-zoom-out.svg';
import { ReactComponent as InfoIcon } from '../../assets/svg/ic-info.svg';

export type IconName = 'zoom-in' | 'zoom-out' | 'info';

export type IProps = { name: IconName } & React.SVGAttributes<SVGElement>;

const Icon: React.FC<IProps> = ({ name, ...props }) => {
  switch (name) {
    case 'zoom-in':
      return <ZoomInIcon {...props} />;
    case 'zoom-out':
      return <ZoomOutIcon {...props} />;
    case 'info':
      return <InfoIcon {...props} />;
    default:
      return null;
  }
};

export default Icon;
