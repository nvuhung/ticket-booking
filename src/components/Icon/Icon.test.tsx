import React from 'react';
import 'jest-styled-components';

import Icon, { IProps } from './Icon';
import { mount, shallow } from '../../test/enzyme';

describe('Icon', () => {
  let props: IProps = (null as any) as IProps;
  let mountedComp: any;

  const wrapper = (bShallow: boolean) => {
    if (!mountedComp) {
      mountedComp = bShallow ? shallow(<Icon {...props} />) : mount(<Icon {...props} />);
    }
    return mountedComp;
  };

  beforeEach(() => {
    props = {
      name: 'zoom-in'
    };
  });

  it('renders without crashing', () => {
    const comp = wrapper(false);
    expect(comp).toMatchSnapshot();
  });

  it('always renders a svg', () => {
    const comp = wrapper(false);
    expect(comp.find('svg').length).toEqual(1);
  });

  describe('when icon name is invalid', () => {
    beforeEach(() => {
      props.name = null as any;
      mountedComp = undefined;
    });

    it('should not render', () => {
      const comp = wrapper(false);
      expect(comp).toBeEmptyRender();
    });
  });
});
