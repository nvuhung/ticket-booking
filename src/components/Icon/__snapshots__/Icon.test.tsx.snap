// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders without crashing 1`] = `
<Icon
  name="zoom-in"
>
  <ForwardRef(SvgIcZoomIn)>
    <svg>
      ic-zoom-in.svg
    </svg>
  </ForwardRef(SvgIcZoomIn)>
</Icon>
`;
